driver-kafka/deploy/ssd-deployment/provision-kafka-aws.tf

```diff
@@ -120,32 +120,34 @@ resource "aws_instance" "zookeeper" {
   }
 }
 
-resource "aws_instance" "kafka" {
+resource "aws_spot_instance_request" "kafka" {
   ami                    = "${var.ami}"
   instance_type          = "${var.instance_types["kafka"]}"
   key_name               = "${aws_key_pair.auth.id}"
   subnet_id              = "${aws_subnet.benchmark_subnet.id}"
   vpc_security_group_ids = ["${aws_security_group.benchmark_security_group.id}"]
   count                  = "${var.num_instances["kafka"]}"
+  wait_for_fulfillment   = true
 
   tags = {
     Name = "kafka-${count.index}"
   }
 }
 
-resource "aws_instance" "client" {
+resource "aws_spot_instance_request" "client" {
```

Comment:
please revert this part about using "spot instances"
IIUC this is not required for the Kafka upgrade